### PR TITLE
Support migration to FP16 optimizer 

### PR DESCRIPTION
**Pull request by tjruwase**
1) Enable initializing fp32 params from non-DeepSpeed checkpoint
2) Avoid repeated gradient norm computation

### CODE REVIEW

File: deepspeed/pt/deepspeed_light.py [hunk 1]
```diff
@@ -164,6 +164,9 @@ def __init__(self,
         self.training_dataloader = self.deepspeed_io(
             training_data) if training_data else None
 
+        # Configure wall clock timer
+        self.timers = SynchronizedWallClockTimer()
```
Comment (eltonzheng) [inline deepspeed/pt/deepspeed_light.py:168]: self.timers already exists, maybe need a rebase?

File: deepspeed/pt/fp16_optimizer.py [hunk 2]
```diff
@@ -314,6 +351,10 @@ def state_dict(self):
         state_dict['clip_grad'] = self.clip_grad
         return state_dict
 
+    def refresh_fp32_params(self):
```
Comment (eltonzheng) [inline deepspeed/pt/fp16_optimizer.py:354]: where is this function called?

File: deepspeed/pt/fp16_optimizer.py [hunk 3]
```diff
@@ -182,9 +205,14 @@ def step(self, closure=None):
 
             self.fp32_groups_flat[i].grad = grads_groups_flat[i]
 
-            norm_groups.append(get_grad_norm(self.fp32_groups_flat, mpu=self.mpu))
+        self.start_timers([COMPUTE_NORM])
+        all_groups_norm = get_grad_norm(self.fp32_groups_flat, mpu=self.mpu)
```
Comment (eltonzheng) [inline deepspeed/pt/fp16_optimizer.py:209]: good catch!

File: deepspeed/pt/fp16_optimizer.py (hunk 2 above)
Comment (tjruwase) [reply deepspeed/pt/fp16_optimizer.py:354]: This will be called from the client function in the case where non-DeepSpeed checkpoint is loaded to initialize the model.
